transport/src/main/java/io/netty/channel/ChannelHandlerAdapter.java

```diff
@@ -64,20 +64,4 @@ public boolean isSharable() {
         }
         return sharable;
     }
-
```

Comment:
The class is needed for the `added` field which is then used by `DefaultChannelPipeline#checkMultiplicity(...)` in conjunction with `isSharable()`.

Reply:
@njhill as @rkapsi said... I am currently trying to come up with something that will remove this need and is not super expensive.

Reply:
@normanmaurer @rkapsi I see what you mean. How about a static `IdentityHashMap` guarded by a R/W lock? We're synchronization on the handler list anyhow everywhere that the check is made. Can be removed when the handler is removed from the pipeline. Or do you think that would entail too much overhead?

Reply:
@njhill I am a bit concerned about the performance impact of this as it would basically be an static synchronized and so all pipelines would act on it :/ I wonder if we maybe just want to allow the user to enable the `Sharable` check via a system property and have it disabled by default. Once the user has tested its handlers in development it should not need to have the check anymore anyway in prod. /cc @rkapsi

Reply:
@normanmaurer my hunch is that it would mostly defeat the purpose of the check if users had to explicitly enable it, since many would not think to. I guess a `ConcurrentHashMap` would probably work too, since I assume that `ChannelHandler`s don't override equals/hashCode (and I can't think of a reason that they would need to).  It's probably not a big deal though, just a shame to need that abstract class just for that reason.

Reply:
@njhill yeah...  let me think a bit more about it